src/Nethermind/Nethermind.Trie/TrieNode.cs

```diff
@@ -761,6 +760,26 @@ void ThrowAlreadySealed()
             }
         }
 
+        /// <summary>
+        /// Method to avoid expensive Stelem_Ref covariant checks
+        /// when setting to object[] array
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private void SetItem(int i, TrieNode node)
+        {
+            int index = IsExtension ? i + 1 : i;
+            var data = _data;
+            if ((uint)index > (uint)data.Length)
+            {
+                // Since we are accessing unsafely we need to do our own
+                // bounds check to ensure is safe; as the Jit won't help us.
+                ThrowIndexOutOfRangeException();
+            }
```

Comment:
I wonder if we could skip the checks in release mode as the index should come from previous logic and be correct (we never had out of range exception on this path)